asts/src/index.ts

```diff
@@ -0,0 +1,37 @@
+import * as ts from 'typescript/lib/tsserverlibrary';
+import * as mm from 'minimatch';
+import * as path from 'path';
+
+let DEFAULT_CONFIG = { include: "assembly/**/*.ts", rootPath: '' };
+let PROJECT: ts.server.Project | null = null;
+
+const create = ({ languageService, project }: ts.server.PluginCreateInfo): ts.LanguageService => {
+  PROJECT = project;
+  return {
+    ...languageService,
+    getCompilerOptionsDiagnostics: () => [],
+      getSyntacticDiagnostics: filter(languageService, "getSyntacticDiagnostics"),
+    getSemanticDiagnostics: filter(languageService, "getSemanticDiagnostics"),
+    getSuggestionDiagnostics: filter(languageService, "getSuggestionDiagnostics"),
+  };
+};
+
+const init = (): ts.server.PluginModule => ({
+  create,
+  onConfigurationChanged: (config) => {
+    DEFAULT_CONFIG = config;
+    if (PROJECT) {
+      PROJECT.refreshDiagnostics();
+    }
+  }
```

Comment:
@jacobsteves I'd appreciate a thorough 🎩 here, since the documentation on this is really scarce so I'm not 100% sure of my implementation, when I tested this it worked correctly though. Basically the test here is to make sure that no TS diagnostics are surfaced in AS sources and that TS sources contain TS diagnostics (and potentially AS diagnostics too, which is a bug that I'll fix in the server as mentioned in the OP). The reason why I'm a bit unsure about this change is that there might be a _temporary_ case in which the diagnostics are mixed, up until the `onConfigurationChanged` callback is triggered and up until the `refreshDiagnostics` method resolves 🤷 , again during my tests it didn't made a difference (it was fast) but I'm clearly biased 😄